filters/canada/british-columbia/vancouver/public-art-json-to-json.js

```diff
@@ -1,25 +1,86 @@
-const filter = function (data, std_lib, stringify, skip_errors) {
+const filter = function (data, std_lib, params) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let validate_params = std_lib.get("validate_params");
+    
+    // validate parameters object
+    params = validate_params(params);
+   
+    // convert JSON data to object form
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
     
+    // define new data and errors array
     let new_data = [];
+    let errors = [];
 
+    // iterate through each data entry
     data.map(d => {
         let item = {};
-        item.name = d.title_of_work;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+        
+        // add name (required)
+        if (!add_required(item, "name", d, d.title_of_work, params.skip_errors, errors)) {
+            skip = true;
+        }
+    
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.geo_point_2d?.lon, params.skip_errors, errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: d.geo_point_2d?.lon, latitude: d.geo_point_2d?.lat };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, d.geo_point_2d?.lat, params.skip_errors, errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
-        new_data.push(item);
+
+        // add optional fields
+
+        // add details sequentially example
+        item.details = {};
+        add_if_not_null(item.details, "description", d.descriptionofwork);
+        add_if_not_null(item.details, "artists_description", d.artistprojectstatement);
+        
+        // reset details for batch example
+        item.details = {};
+
+        // add details batch example
+        item.details.description = d.descriptionofwork;
+        item.details.artists_description = d.artistprojectstatement;
+        item.details = remove_if_null(item.details);
+        
+        // check for and remove empty details object
+        if (Object.keys(item.details).length <= 0) {
```

Comment:
this could be a good function to go in the standard library to avoid redundancy

Reply:
agreed, ill add it in